### PR TITLE
Refactor AudioStream methods to provide safe defaults when stream is closed

### PR DESCRIPTION
**Pull request by dturner**
Questions: 

- should `isXRunCountSupported` return false for OpenSL ES?

### CODE REVIEW

**Comment by philburk** [comment]
isXRunCountSupported() should return false when using a callback, and true when using blocking read/write.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
+add_subdirectory(${OBOE_DIR} ./oboe-bin)
+include_directories(${OBOE_DIR}/include)
+
+# Build the test binary
+add_executable(testOboe
+        testUtilities.cpp
+        testStreamClosedMethods.cpp
+        testXRunBehaviour.cpp
+        )
+target_link_libraries(testOboe gtest oboe)
```
Comment (philburk) [inline tests/CMakeLists.txt]: add newline

File: src/aaudio/AudioStreamAAudio.cpp [hunk 2]
```diff
@@ -268,7 +271,7 @@ Result AudioStreamAAudio::requestStart() {
     if (stream != nullptr) {
         return static_cast<Result>(mLibLoader->stream_requestStart(stream));
     } else {
-        return Result::ErrorNull;
+        return Result::ErrorClosed;
```
[the remark this answers is not in the document]
Comment (dturner) [reply src/aaudio/AudioStreamAAudio.cpp]: Ah, good spot. It was wrong before and I made it wronger :)

File: include/oboe/Definitions.h [hunk 3]
```diff
@@ -96,6 +96,7 @@ namespace oboe {
         ErrorOutOfRange = AAUDIO_ERROR_OUT_OF_RANGE,
         ErrorNoService = AAUDIO_ERROR_NO_SERVICE,
         ErrorInvalidRate = AAUDIO_ERROR_INVALID_RATE,
+        ErrorClosed,
```
[the remark this answers is not in the document]
Comment (dturner) [reply include/oboe/Definitions.h]: Hmm, good point. We shouldn't really be casting from `aaudio_result_t` to `Result` but we do rely on it here: https://github.com/google/oboe/blob/1b4b1af24e9b46606860e6f1db1190937dfd0561/src/aaudio/AudioStreamAAudio.cpp#L204. 

Stupid question but how does one set aside values for the future? Is it simply `ErrorUnused1, ErrorUnused2` etc?

[the remark this answers is not in the document]
Comment (dturner) [reply include/oboe/Definitions.h]: Done. Added 10 reserved values for future use.

File: src/aaudio/AudioStreamAAudio.cpp [hunk 4]
```diff
@@ -176,6 +176,8 @@ Result AudioStreamAAudio::open() {
     mPerformanceMode = static_cast<PerformanceMode>(
             mLibLoader->stream_getPerformanceMode(mAAudioStream));
     mBufferCapacityInFrames = mLibLoader->stream_getBufferCapacity(mAAudioStream);
+    mBufferSizeInFrames = mLibLoader->stream_getBufferSize(mAAudioStream);
+
```
[the remark this answers is not in the document]
Comment (dturner) [reply src/aaudio/AudioStreamAAudio.cpp]: These are all cached during `open` - what's your point?

File: src/aaudio/AudioStreamAAudio.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (philburk) [reply src/aaudio/AudioStreamAAudio.cpp]: Oops. Never mind. I misread the diff display.

**Pull request merged**